triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerLogContext.scala

```diff
@@ -51,24 +51,44 @@ final class TriggerLogContext private (
 
   import ToLoggingContext._
 
+  private[this] var callback: (String, TriggerLogContext) => Unit = (_, _) => ()
+
+  private[trigger] def setCallback(callback: (String, TriggerLogContext) => Unit): Unit = {
+    this.callback = callback
+  }
+
   def enrichTriggerContext[A](
       additionalEntries: (String, LoggingValue)*
   )(f: TriggerLogContext => A): A = {
-    f(new TriggerLogContext(loggingContext, entries ++ additionalEntries, span))
+    val context = new TriggerLogContext(loggingContext, entries ++ additionalEntries, span)
+
+    context.setCallback(callback)
```

Comment:
We need to ensure all newly created logging contexts have the current callback defined upon them

Reply:
if we always need it, wouldn't it make sense to make `callback` a field of the `TriggerLogContext` constructor? maybe that's me thinking in Haskell though

Reply:
Good question - and thanks for getting myself to focus on this decision. I'd originally done this as it was a quick "hack" to achieve a means (with future PRs). 

I've refactored the `TriggerLogContext` to now always take this callback argument. When I eventually define code to use this callback, I'll have to work a little harder to define a logging context (so that a root trigger logging context may be defined).